File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02636/p02636_num7_parsed.py
MOD = 998244353

def count_strings(S):
    n = len(S)
    counts = [[0] * n for _ in range(n)]
    
    for i in range(n - 2):
        if S[i] == S[i + 1]:
            counts[i][i + 2] = 1
    
    for l in range(4, n + 1, 2):
        for i in range(n - l + 1):
            for j in range(i + 2, i + l, 2):
                if S[i] == S[j]:
                    counts[i][i + l] += counts[i + 2][j] * counts[j + 2][i + l]
                    counts[i][i + l] %= MOD
    
    return counts[0][n]

S = input()
result = count_strings(S)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02636/p02636_num7_parsed.cpp
#include <iostream>
#include <vector>

const int MOD = 998244353;

int main() {
    std::string S;
    std::cin >> S;

    int N = S.length();
    
    std::vector<int> dp(N + 1, 0);
    dp[0] = 1;
    dp[1] = 0;
    dp[2] = 1;

    for (int i = 3; i <= N; i++) {
        dp[i] = (dp[i - 1] + dp[i - 3]) % MOD;
    }
    
    int count = 0;
    for (int i = 0; i < N; i++) {
        if (S[i] == '1') {
            count++;
        }
    }

    int result = dp[count];

    std::cout << result << std::endl;

    return 0;
}
